Content.Server/_CD/Traits/LayEmoteSystem.cs

```diff
@@ -0,0 +1,84 @@
+using Content.Server.Chat.Systems;
+using Content.Shared.Body.Systems;
+using Content.Shared.Buckle.Components;
+using Content.Shared.Humanoid;
+using Content.Shared.Mobs;
+using Content.Shared.Movement.Systems;
+using Content.Shared.Standing;
+
+namespace Content.Server._CD.Traits;
+
+public sealed class LayEmoteSystem : EntitySystem
+{
+    [Dependency] private readonly MovementSpeedModifierSystem _modifier = default!;
+    [Dependency] private readonly StandingStateSystem _standingSystem = default!;
+    [Dependency] private readonly SharedBodySystem _bodySystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LayEmoteComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<LayEmoteComponent, BuckleChangeEvent>(OnBuckleChange);
+        SubscribeLocalEvent<LayEmoteComponent, MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<LayEmoteComponent, EmoteEvent>(OnEmote);
+        SubscribeLocalEvent<LayEmoteComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovementSpeedModifiers);
+    }
+
+    // On shutdown, make sure people are standing and reset movement speed
+    private void OnShutdown(EntityUid uid, LayEmoteComponent component, ComponentShutdown args)
+    {
+        _standingSystem.Stand(uid);
+        _bodySystem.UpdateMovementSpeed(uid);
+    }
+
+    // If buckled, make sure someone is standing. Unbuckling while laying down should keep someone laying down and vice versa.
+    private void OnBuckleChange(EntityUid uid, LayEmoteComponent component, ref BuckleChangeEvent args)
+    {
+        if (args.Buckling && component.Laying)
+            _standingSystem.Stand(args.BuckledEntity);
+
+        if (!args.Buckling && component.Laying)
+            _standingSystem.Down(args.BuckledEntity);
+    }
+
+    private void OnMobStateChanged(EntityUid uid, LayEmoteComponent component, MobStateChangedEvent args)
+    {
+        // Hoping this should work fine as going crit - dead or dead - crit shouldn't matter, and crit - alive would stand you up anyways.
+        component.Laying = false;
```

Comment:
i imagine this could possibly troll your movement speed, add a SetLaying function that 1. checks if arg isnt the same as Laying and 2. does the Stand Down update movespeed etc
then use that everywhere

then here call SetLaying(ent, false);